packages/react-scripts/template/README.md

```diff
@@ -803,6 +804,76 @@ In the future we plan to integrate it into Create React App even more closely.
 
 To learn more about Flow, check out [its documentation](https://flow.org/).
 
+## Adding TypeScript
+
+TypeScript is a typed superset of JavaScript that compiles to plain JavaScript.
+
+Recent versions of [TypeScript](https://www.typescriptlang.org/) work with Create React App projects out of the box thanks to Babel 7. Beware that Babel 7 TypeScript does not allow some features of TypeScript such as constant enum and namespaces.
+
+To add TypeScript to a Create React App project, follow these steps:
+
+1. Run `npm install --dev typescript @types/react @types/react-dom @types/jest` (or `yarn add --dev typescript @types/react @types/react-dom @types/jest`).
+2. Rename the `.js` files you want to convert. Use `.tsx` if they use JSX or `.ts` if not (e.g. `git mv src/index.js src/index.tsx`).
+
+3. Create a `tsconfig.json` file at the root directory with the following content:
+
+```json
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "module": "esnext",
+    "moduleResolution": "node",
+    "lib": ["esnext", "dom"],
+    "allowJs": true,
+    "allowSyntheticDefaultImports": true,
+    "emitDecoratorMetadata": false,
+    "esModuleInterop": true,
+    "experimentalDecorators": false,
```

Comment:
Why `"experimentalDecorators": false`? Will it fail with `true`?

Reply:
CRA doesn’t support decorators

Reply:
Oh. Although TS could compile them down?

Atm. I'm using decorators in my create-react-app-typescript project. Works nice.

Reply:
AFAIK CRA 2.0 doesn’t support it by default. But yes you can change this config to any config you want.

Reply:
I thought about removing the `"experimentalDecorators": false` (because it's the TS default anyways). But let's see if it works and decide later. Thanks for your reply!